lalrpop/src/grammar/parse_tree.rs

```diff
@@ -118,6 +119,7 @@ pub struct InternToken {
     /// Set of `r"foo"` and `"foo"` literals extracted from the
     /// grammar. Sorted by order of increasing precedence.
     pub literals: Vec<TerminalLiteral>,
+    pub match_to_user_name_map: Option<Map<TerminalString, TerminalString>>,
```

Comment:
In general, I think it would work out better if we merged this with the `literals` Vector into a vector of pairs. This would also force us to consider, each time we iterate, whether we want the *actual literal* that is being matched or the *user literal*, which seems likely to prevent bugs.

(Even if we kept it this way, I personally would not use an `Option<Map<..>>` but just a `Map<..>`, and I'd say that if there is no `match` then either it's an empty map, or we populate it with a `X -> X` mappings.)